File: Solutions in Java/RomanToInteger.java
/**
 * @author : Anmol Agrawal
 */
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> hm = new HashMap<>();
        hm.put('I', 1);
        hm.put('V', 5);
        hm.put('X', 10);
        hm.put('L', 50);
        hm.put('C', 100);
        hm.put('D', 500);
        hm.put('M', 1000);
        int ans = 0;
        int previous = 0;
        for(int i = 0; i < s.length(); i++){
            int value = hm.get(s.charAt(i));
            ans += value;
            if(previous < value){
                ans -= previous;
                ans -= previous;
            }
            previous = value;
        }
        return ans;
    }
}

File: Solutions in Javascript/RomanToInteger.js
/**
 * @author : Anmol Agrawal
 */
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    let map = new Map();
    map.set('I', 1);
    map.set('V', 5);
    map.set('X', 10);
    map.set('L', 50);
    map.set('C', 100);
    map.set('D', 500);
    map.set('M', 1000);
    let ans = 0;
    let previous = 0;
    for(let i = 0; i < s.length; i++){
        let value = map.get(s.charAt(i));
        ans += value;
        if(previous < value){
            ans -= previous;
            ans -= previous;
        }
        previous = value;
    }
    return ans;
};